src/figuresClasses.ts

```diff
@@ -1,19 +1,70 @@
-export interface Figure {
-
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
 
-export class Triangle {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = Shape.Triangle;
+  }
 
+  getArea(): number {
+    const halfP = (this.a + this.b + this.c) / 2;
+
+    return Math.sqrt(halfP * (halfP - this.a) * (halfP - this.b)
+    * (halfP - this.c));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape;
+
+  constructor(public color: Color, public radius: number) {
+    this.shape = Shape.Circle;
+  }
 
+  getArea(): number {
+    return 3.14 * (this.radius ** 2);
```

Comment:
JS has Math.PI constant for more accurate calculations.

Reply:
Test didn't work with Math.PI (that was the first thing i tried 😄), i guess test were developed with exactly 3.14.

Reply:
🤕